crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -2529,6 +2529,23 @@ impl<F: Clone>
 
         let response_router_data = response;
 
+        if let (true, true, Some(payment_method_id)) = (
```

Comment:
move to domain and call the domain function in core flow instead adding this in trackers